### DISCUSSION TITLE
ESPHome support for Venta AH510 and others

### DISCUSSION THREAD

**Discussion by helgek** [Michsior14/ha-venta#39] [General]
Hi, 

I just finished writing this post https://community.home-assistant.io/t/venta-lw-controller/189608/6?u=helge_k where I wanted to kick off the search for supporters of an ESPHome based configuration to replace the original firmware of ESP32 based Venta devices and then I found your project and realized that with you're already supporting the latest devices with the Rest API approach.

I would be more than happy if you might be also interested in looking into developing an ESPHome config for the Venta devices. I'd be more than happy to also contribute as much as possible but with some technical things I will certainly need help since my knowledge is sometimes not sufficient, especially when it comes to more advanced electronic stuff and understanding of the whole communication on a PCB. Through self-studies I've come pretty far I'd say also this reaches its limits.

Like I've also stated in the linked post I've already downloaded the original firmware and I also did photo documentation of the PCB which I both would be happy to share of course.

Best,
Helge

**Comment by Michsior14** [comment]
Hey, I don't have that much of experience in PCBs as well. Hopefully someone with greater knowledge will jump in here and help if you are stuck. If you create something give me an update I will test it as well and link to your solution.

**Reply by helgek** [reply]
I definitely will let you know. Actually when I started with the ESPHome idea I hadn't been aware of your solution. Now that I'm aware of it the pressure to go the ESPHome route is not that big anymore. Nevertheless I think it would be nice if it would happen one day. It would certainly once again optimize the control experience of the device, both performance-wise as well as regarding some additional control features like turning off the display completely if desired.

In the web interface I saw that you can upload a binary file for OTA. Maybe one could even replace the original firmware without the need the flash through the PCB. I wasn't in experimental mode today though after having assembled the device again although theoretically it shouldn't be a problem to switch forth and back between original firmware and ESPHome firmware.

<img width="1026" alt="image" src="https://github.com/Michsior14/ha-venta/assets/6280991/b12568f0-7b04-4fd0-8f32-4be10c03f68c">

This is btw the untouched original firmware of my AH510 which I downloaded before I set the wifi up for my purpose:

[venta.bin.zip](https://github.com/Michsior14/ha-venta/files/14915218/venta.bin.zip)
